src/accelerator.rs

```diff
@@ -0,0 +1,89 @@
+use drivers::Driver;
+use motions::Motion;
+
+fn clamp(n: isize, min: isize, max: isize) -> isize {
+    if n < min {
+        return min;
+    }
+    if n > max {
+        return max;
+    }
+    n
+}
+
+fn execute<D: Driver>(driver: &mut D, motions: &Vec<Motion>, mut start: isize, mut finish: isize) {
```

Comment:
Should this be private to denote it should not be used outside of this module/file? Or at least marked as private with an underscore (`_execute`).